### PR TITLE
1400: Activate card confirmation mail per region

### PR DESCRIPTION
**Pull request by f1sh1918**
### Short description

As a region i may not want to make the card earlier usable and not send an activation mail.

### Proposed changes

- adjust activated for application to region specific settings and add checkbox for activate conformation mail
- also adjusted the query and update functions to not have x single functions for each setting
- send only card confirmation mail if its activated
- set the confirmationFeatureFlag true
- add unit test
- added jest-dom and types to enable checkbox checking in tests

### Resolved issues

Fixes: #1400 

### Testing
- run db migration (dev only)
- login as region admin (bavaria)
- go to region
- set checkbox activate confirmation mail
- create card from application
- toast 'Benachrichtigung wurde gesendet'  should be shown
- mail should be sent
- uncheck the checkbox in region settings
- create card from application
- mail shouldn't be send

**Notes:** Many changes are just because region data class has changed

### CODE REVIEW

**Review by ztefanie — CHANGES_REQUESTED** [review]
- Tested settings on Bayern, the settings are only visibale for the role region-admin. not for other roles or nuernberg. I expect this is the wanted behaviour.

- Route is protected, but when sending the request without authorization I get an 401 as expected. When I send a request with the authorization for the wrong region, i get an 200, with the message "Exception while fetching data (/result) : Insufficient access right" and a stack trace. I think this should als return a 401, without any additional data.

- Sending mails does not work for me. (It did previously and it also works on main). The PDF is created but an error is shown and no email is sent.) See the attached video:

https://github.com/digitalfabrik/entitlementcard/assets/18463686/af64c142-47ee-403b-9d88-e5ab10102087

File: administration/src/bp-modules/cards/hooks/useCardGenerator.ts [hunk 1]
```diff
@@ -89,7 +89,7 @@ const useCardGenerator = (region: Region) => {
 
       cardBlueprints.forEach(cardBlueprint => new ActivityLog(cardBlueprint).saveToSessionStorage())
       downloadDataUri(pdfDataUri, 'berechtigungskarten.pdf')
-      if (projectConfig.cardCreationConfirmationMailEnabled) {
+      if (region.activatedForCardConfirmationMail) {
```
[the remark this answers is not in the document]
Comment (f1sh1918) [reply administration/src/bp-modules/cards/hooks/useCardGenerator.ts]: indeed. for nuernberg in general there is no "region verwalten" function. All region verwalten functions are connected to the applications which do not exist for nuernberg.
For bavaria region admins can adjust their settings individual

File: backend/src/main/kotlin/app/ehrenamtskarte/backend/migration/migrations/V0011_AddRegionCardConfirmationMailActivation.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ * Add activatedForCardConfirmationMail column to regions.
+ * Set default to false.
+ */
+@Suppress("ClassName")
+internal class V0011_AddRegionCardConfirmationMailActivation() : Migration() {
+    override val migrate: Statement = {
+        exec(
+            """
+            ALTER TABLE regions 
+            ADD "activatedForCardConfirmationMail" BOOLEAN NOT NULL DEFAULT false;
```
[the remark this answers is not in the document]
Comment (f1sh1918) [reply backend/src/main/kotlin/app/ehrenamtskarte/backend/migration/migrations/V0011_AddRegionCardConfirmationMailActivation.kt:16]: yes as daniel told me, every region should opt-in this to activate

**Comment by f1sh1918** [comment]
> * Tested settings on Bayern, the settings are only visibale for the role region-admin. not for other roles or nuernberg. I expect this is the wanted behaviour.
> * Route is protected, but when sending the request without authorization I get an 401 as expected. When I send a request with the authorization for the wrong region, i get an 200, with the message "Exception while fetching data (/result) : Insufficient access right" and a stack trace. I think this should als return a 401, without any additional data.
> * Sending mails does not work for me. (It did previously and it also works on main). The PDF is created but an error is shown and no email is sent.) See the attached video:
> 
>  1400.mp4

1. is correct
2. i think a 403 should be correct and to throw a forbidden exception
3. i fixed that by surround it by a transaction

**Review by ztefanie — APPROVED** [review]
Not tested again. Looks good.

Using 403 Forbidden leads to exposing which resources exist even if not authorized, that why i suggested 401, but I am completely fine with using 401, just an explanation.

**Pull request merged**